src/common/components/repositories-home/index.js

```diff
@@ -89,10 +89,11 @@ class RepositoriesHome extends Component {
 
   render() {
     const { snaps } = this.props;
+    const hasSnaps = (snaps.snaps && Object.keys(snaps.snaps).length > 0);
     // show spinner until we know if user has any enabled repos
-    return (snaps.success || snaps.error)
-        ? this.renderRepositoriesList()
-        : this.renderSpinner();
+    return (snaps.isFetching && !hasSnaps)
+      ? this.renderSpinner()
+      : this.renderRepositoriesList();
```

Comment:
I've changed the criteria for showing the repositories list to avoid explicitly overriding the "isFetching" status.

`snaps.success` is not a reliable criteria, because it changes whenever we poll the service for updated snap information. Instead, I've used the `isFetching` status coupled with the existence of snaps.